analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.cs

```diff
@@ -269,5 +270,30 @@ public void Branching_VisitedProgramState_IsImmutable()
             validator.ValidateExitReachCount(2);
             captured.Should().OnlyContain(x => x.Value.HasConstraint(BoolConstraint.True) == x.ExpectedHasTrueConstraint);
         }
+
+        [TestMethod]
+        public void Branching_VisitedSymbolicValue_IsImmutable()
+        {
+            const string code = @"
+var value = true;
+if (boolParameter)
+{
+    value.ToString();
+    Tag(""ToString"", value);
+}
+else
+{
+    value.GetHashCode();    // Another invocation to have same instruction count in both branches
+    Tag(""GetHashCode"", value);
+}";
+            var postProcess = new PostProcessTestCheck(x =>
+                x.Operation.Instance is IInvocationOperation invocation && invocation.TargetMethod.Name == "ToString"
+                    ? x.SetSymbolConstraint(invocation.Instance.TrackedSymbol(), TestConstraint.First)
+                    : x.State);
+            var validator = SETestContext.CreateCS(code, new BoolTestCheck(), postProcess).Validator;
```

Comment:
Why do we need the `new BoolTestCheck()`? We don't assert any True/False constraints.

Reply:
It can be removed (and forces a little change in assertions)